src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.Calculator/CalculateHelper.cs

```diff
@@ -60,6 +61,13 @@ public static string FixHumanMultiplicationExpressions(string input)
             return output;
         }
 
+        private static string CheckScientificNotation(string input)
+        {
+            var p = @"(-?(\d+(\.\d*)?)|-?(\.\d+))[eE](-?\d+\.*\d*)";
```

Comment:
```suggestion
            var p = @"(-?\d+(\.\d*)?)[eE](-?\d+(\.\d*)?)";
```
I've simplified the regex:
- Remove unnecessary capturing groups and consolidate mantissa section
- Fix the exponent section to be of the same form as the mantissa
  - Does not allow unlimited decimal points anymore

Reply:
The simplified expression doesn't capture if anything that doesn't start with a digit, so if you had something like `.5e2`, it gets translated to `.(5 * 10^(2))`, which isn't right. That being said, if it's not valid to do something like `.5e2`, I like the new expression better

It's why that `|` is in there, since you could theoretically start with a decimal, though I do think that's unlikely

Reply:
Oh, I see what you were doing now.
`2e.5` probably shouldn't work since exponential form is typically to a whole number, so that makes sense why you had the two sides different.

Reply:
What do you mean with unlimited decimal points? What was being accepted before that now is not?

Reply:
Ah, I see. The suggestions were declined.

Reply:
The unlimited decimals is accepted because of the `\•*` after the `e`, should be `\.?` instead. Otherwise yeah, I wanted to leave the expression as is since not everyone likes to put digits before decimal places